octavia-cli/octavia_cli/headers.py

```diff
@@ -0,0 +1,114 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+from dataclasses import dataclass
+from typing import Optional, List
+
+import click
+import yaml
+
+from .apply.yaml_loaders import EnvVarLoader
+
+
+API_HEADERS_YAML_FILE_INSTRUCTION = """
+headers:
+  - name: Authorization
+    value: Basic dXNlcjpwYXNzd29yZA==
+  - name: Content-Type
+    value: application/json
+""".strip()
+
+
+@dataclass
+class ApplicationHeader:
+    name: Optional[str]
+    value: Optional[str]
+
+    @property
+    def is_valid(self):
+        return all([self.name, self.value])
+
+    def clean(self):
+        return ApplicationHeader(
+            name=self.name.strip(),
+            value=self.value.strip()
+        )
+
+
+@dataclass
+class ApiHeadersResult:
+    raw_length: int
+    validated_length: int
+    valid_application_headers: List[ApplicationHeader]
+
+    def is_valid(self):
+        return self.raw_length == self.validated_length
+
+
+def read_yaml_header_configuration(header_configuration_path: str) -> ApiHeadersResult:
+    with open(header_configuration_path) as file:
+        content = yaml.load(file, EnvVarLoader)
+
+    headers = content.get("headers", None)
+
+    if not headers:
+        raise AttributeError("""
+        Please provide valid configuration file, like below
+        headers:
+          - name: Authorization
+            value: Basic dXNlcjpwYXNzd29yZA==
+          - name: Content-Type
+            value: application/json
+        """)
+
+    application_headers = [
+        ApplicationHeader(header_pair.get("name", None), header_pair.get("value")) for header_pair in headers
+    ]
+
+    valid_headers = [
+        app_header.clean() for app_header in application_headers if app_header.is_valid
+    ]
+
+    return ApiHeadersResult(
+        raw_length=len(application_headers),
+        validated_length=len(valid_headers),
+        valid_application_headers=valid_headers
+    )
+
+
+def get_argument_based_headers(api_headers: List[str]) -> ApiHeadersResult:
```

Comment:
I'd prefer `api_headers` to be a list of tuples built from the multi-value option of Click https://click.palletsprojects.com/en/8.1.x/options/#multi-value-options

It will allow to avoid the `.split(":")`

Reply:
I unresolve the conversation because you did not change the code or replied to my suggestion